src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -246,6 +243,14 @@ protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWa
             DependencyResolver = serviceLocator.ResolveType<IDependencyResolver>();
             _dispatcherService = DependencyResolver.Resolve<IDispatcherService>();
 #endif
+            var type = GetType();
+
+            serviceLocator.RegisterTypeIfNotYetRegisteredWithTag(typeof(IObjectIdGenerator<int>), typeof(IntegerObjectIdGenerator<>).MakeGenericType(type), type);
+
+            _objectIdGenerator = serviceLocator.ResolveType<IObjectIdGenerator<int>>(type);
+            UniqueIdentifier = _objectIdGenerator.GetUniqueIdentifier();
```

Comment:
I wrote in the description of this feature that for me the problem is the inability to override this behavior. Now I can implement my own IObjectIdGenerator<int>, but I need to implement reusing in GetUniqueIdentifier() method anyway. It's OK for me, but what if we add argument "reuse identifier" in this ctor?

Reply:
I'm still thinking how introduce it without adding as VM constructor parameter. At this point you can just implement your own IObjectIdGenerator that always reuse the identifier by default.

Reply:
Maybe make sense remove the 'reuse' overload at all?

Reply:
Then what about "ReleaseIdentifier"? It will be empty?

Reply:
I have this idea: if we can register "reuse" flag per type, then we don't need any additional parameters, and also we can avoid the filling (and even allocation) of the queue when "reuse" has false value.

Reply:
I pushed a solution but is a breaking change because add it an optional parameter to ViewModelBase constructor